src/services/item/plugins/importExport/itemExportRequest.service.ts

```diff
@@ -0,0 +1,172 @@
+import fs from 'fs';
+import path from 'path';
+import { pipeline } from 'stream/promises';
+import { singleton } from 'tsyringe';
+import { ZipFile } from 'yazl';
+
+import { UnionOfConst } from '@graasp/sdk';
+
+import { DBConnection } from '../../../../drizzle/db';
+import { type ItemRaw, MinimalAccount } from '../../../../drizzle/types';
+import { TRANSLATIONS } from '../../../../langs/constants';
+import { BaseLogger } from '../../../../logger';
+import { MailBuilder } from '../../../../plugins/mailer/builder';
+import { MailerService } from '../../../../plugins/mailer/mailer.service';
+import { TMP_FOLDER } from '../../../../utils/config';
+import FileService from '../../../file/file.service';
+import { MemberService } from '../../../member/member.service';
+import { ItemRequestExportRepository } from './itemExportRequest.repository';
+
+const EXPORT_ITEM_EXPIRATION_DAYS = 7;
+const EXPORT_ITEM_EXPIRATION = 3600 * 24 * EXPORT_ITEM_EXPIRATION_DAYS; // max value: one week
+
+export const ItemRequestExportType = {
+  Raw: 'raw',
+  Graasp: 'graasp',
+} as const;
+type ItemRequestExportTypeOptions = UnionOfConst<typeof ItemRequestExportType>;
+
+@singleton()
+export class ItemExportRequestService {
+  private readonly fileService: FileService;
+  private readonly itemRequestExportRepository: ItemRequestExportRepository;
+  private readonly mailerService: MailerService;
+  private readonly memberService: MemberService;
+  private readonly logger: BaseLogger;
+
+  constructor(
+    fileService: FileService,
+    itemRequestExportRepository: ItemRequestExportRepository,
+    mailerService: MailerService,
+    memberService: MemberService,
+    logger: BaseLogger,
+  ) {
+    this.fileService = fileService;
+    this.itemRequestExportRepository = itemRequestExportRepository;
+    this.mailerService = mailerService;
+    this.memberService = memberService;
+    this.logger = logger;
+  }
+
+  /**
+   * path where an export request is saved on the file storage
+   * @param requestId
+   * @returns
+   */
+  private buildExportPath(requestId: string) {
+    return `item-export/${requestId}`;
+  }
+
+  /**
+   * Upload archive and send email with the download link
+   * @param dbConnection
+   * @param actor
+   * @param item
+   * @param archive zip content of the item to be uploaded
+   */
+  async uploadAndSendDownloadLink(
+    dbConnection: DBConnection,
+    actor: MinimalAccount,
+    item: ItemRaw,
+    archive: ZipFile,
+    type: ItemRequestExportTypeOptions,
+  ) {
+    // upload zip
+    const filepath = await this.uploadZip(dbConnection, actor, item, archive, type);
+
+    // send email
+    const memberWithEmail = (await this.memberService.get(dbConnection, actor.id)).toMemberInfo();
```

Comment:
Split this instruction into two lines. The parenthesis make it likely to swap the order of await and `.toMemberInfo()` making it error-prone.
```suggestion
    const fullMember = await this.memberService.get(dbConnection, actor.id);
    const memberWithEmail = fullMember.toMemberInfo();
```